app/Http/Controllers/AuctionController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\AuctionRequest;
+use App\Http\Requests\UpdateAuctionRequest;
+use App\Models\auction;
+use App\Notifications\TelegramNotification;
+use App\Services\AuctionService;
+use App\Services\CategoryService;
+use App\Services\ProductService;
+use ArgumentCountError;
+use Exception;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Notification;
+use Illuminate\Validation\ValidationException;
+
+class AuctionController extends Controller
+{
+    private $auctionService;
+    private $productService;
+    private $categoryService;
+
+    public function __construct(AuctionService $auctionService, ProductService $productService, CategoryService $categoryService)
+    {
+        $this->auctionService = $auctionService;
+        $this->productService = $productService;
+        $this->categoryService = $categoryService;
+    }
+    public function index()
+    {
+        $auctions = $this->auctionService->getAll()->filter()->paginate(10);
+        // dd($auctions);
+        foreach ($auctions as $auction) {
+            $status = $auction->setStatus();
+            if ($auction->status != $status)
+                $this->auctionService->update($auction, ['status' => $status]);
+        }
+        $categories = $this->categoryService->getAll()->get();
+        return view("auctions.index", compact("auctions", "categories"));
+    }
+
+    public function create()
+    {
```

Comment:
if(!$this->authorize("create", Auction::class))  abort();
 $products = $this->productService->getMyProduct();
            return view("auctions.create", compact("products"));